tests/integration/cmd_link_unlink_test.go

```diff
@@ -38,109 +37,58 @@ var _ = Describe("odo link and unlink command tests", func() {
 		})
 	})
 
-	/*
-		Context("When link between components using wrong port", func() {
-			JustBeforeEach(func() {
-				frontendContext = helper.CreateNewContext()
-				backendContext = helper.CreateNewContext()
-			})
-			JustAfterEach(func() {
-				helper.DeleteDir(frontendContext)
-				helper.DeleteDir(backendContext)
-			})
-			It("should fail", func() {
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "--s2i", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				helper.CopyExample(filepath.Join("source", "python"), backendContext)
-				helper.Cmd("odo", "create", "--s2i", "python", "backend", "--context", backendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-				stdErr := helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "1234").ShouldFail().Err()
-				Expect(stdErr).To(ContainSubstring("Unable to properly link to component backend using port 1234"))
-			})
+	Context("When handling link/unlink between components", func() {
+		JustBeforeEach(func() {
+			frontendContext = helper.CreateNewContext()
+			backendContext = helper.CreateNewContext()
+		})
+		JustAfterEach(func() {
+			helper.DeleteDir(frontendContext)
+			helper.DeleteDir(backendContext)
+		})
+		It("should link the frontend application to the backend and then unlink successfully", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
+			helper.Cmd("odo", "create", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
```

Comment:
Do you think there's a scope for extracting few things like this one into `JustBeforeEach`? I see it's repeating twice.

Reply:
> I find the code deviating from your once proposed idea of making sure that It has only the code/command that we're checking. I see some setup happening in the It part. WDYT?

That is because I have not refactored it yet. I just cleaned it up. I was waiting for the reviewers to approve the clean-up before I start to work on the refactoring part. Now that I do have approval, I'll get to it.

Reply:
It's tests PR. Most of the time, the only approval we're looking for is CI to go green. Please don't leave anything on the basis of "approval". Check with someone explicitly instead.